app/controllers/report/learner_controller.rb

```diff
@@ -81,45 +81,33 @@ def setup
     if params[:commit] =~ /update learners/i
       update_learners
     elsif params[:commit] == @button_texts[:logs_query]
-      redirect_to learner_logs_query_path(request.GET.except(:commit))
+      redirect_to learner_logs_query_path(request.GET.merge({:commit => true}))
```

Comment:
I didn't make a comment because I didn't understand the original code well enough -- or rather, I understood what it did fine, but it felt like there must be a simpler way to do it, so I couldn't justify why the original code was the way it was.

Both the `/report/learner` and `/report/learner/logs_query` routes use this same page. Both use the same `setup` function. In the `setup` function, if it sees that it has the `commit= logs_query` param, it will redirect to `/report/learner/logs_query`. In order not to create an infinite redirect loop, it strips the `commit= logs_query` param.

In my code, unlike the original, I wanted to only make a query for the learners when the `commit` param was there -- i.e. when the user had clicked a button. But the way this was originally, when the `commit= logs_query` param was stripped we wouldn't make a query for the learners.

This change keeps some value for the `commit` query, while preventing the infinite redirect that whoever wrote this originally wanted to avoid.

I've now added a simple comment saying that.